### PR TITLE
fixed some clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
Mostly about readability, but also removes a few needless `format!` and `vec!` calls, so it might even marginally improve performance. :smile:

### CODE REVIEW

File: plugin/src/compiler.rs [hunk 1]
```diff
@@ -272,14 +272,12 @@ fn compile_op(ecx: &ExtCtxt, buffer: &mut StmtBuffer, op: Ident, prefixes: Vec<I
...
-                rex_w = true;
-            } else if op_size != Size::DWORD {
-                panic!("bad formatting data");
-            }
+        } else if op_size == Size::WORD {
+            pref_size = true;
+        } else if op_size == Size::QWORD {
+            rex_w = true;
+        } else if op_size != Size::DWORD {
+            panic!("bad formatting data");
```
Comment (CensoredUsername) [inline plugin/src/compiler.rs]: I'd like to keep this part within the else block to be consistent with the previous blocks. A better formatting solution would probably be to just match on op_size for each size encoding flag, it'd be shorter and more clear.

**Comment by llogiq** [comment]
I backed out that change. You'll know best how to improve the formatting.

**Comment by CensoredUsername** [comment]
Thanks for the work, I'll merge it once I'm done with current update.

And I should probably start using clippy.

**Comment by llogiq** [comment]
You're welcome.

**Pull request merged**